jina/executors/indexers/query/__init__.py

```diff
@@ -0,0 +1,11 @@
+from jina.executors.indexers import BaseIndexer
+
+
+class BaseQueryIndexer(BaseIndexer):
+    """An indexer only for querying. It only reads once (at creation time)"""
```

Comment:
If this is only loaded once at creation time, shouldn't `load_dump` be a `private` function, and be the `constructor` detecting if it should be loaded from `dump`?

Reply:
I see this being the pattern, so I think there is no need to add this extra class just to add a `private method`.

Another thing is if there is a pattern on how the existence of `dump` is checked and initialized from it

Reply:
Hm yes we should add a mechanism to ensure it's not called twice

Right now: it would fail since it would call `add` and that's been disabled after the first call

Alternatives:

- private method only called internally
- delete it after it's used once (what we do with `add` now)

Reply:
I would still keep a Base class to have a hierarchy